ocaml/xapi/cli_operations.ml

```diff
@@ -662,11 +663,21 @@ let make_param_funs getall getallrecs getbyuuid record class_name def_filters de
           match field.set_in_map with
           | Some set_in_map -> set_in_map key v
           | None ->
-            let add_to_map = match field.add_to_map with Some f -> f | None -> failwith ("Map field '"^s^"' is read-only.") in
-            let remove_from_map = match field.remove_from_map with Some f -> f | None -> failwith (Printf.sprintf "Records broken (field %s)" s) in
-            let map = get_map () in
-            if List.mem_assoc key map then remove_from_map key;
-            add_to_map key v
+            (* If set_map is present then accumulate all (key, value) pairs. *)
+            match field.set_map with
+            | Some set_map ->
+              if not (Hashtbl.mem set_map_params set_map) then
+                Hashtbl.add set_map_params set_map [(key, v)]
+              else begin
+                let existing_params = Hashtbl.find set_map_params set_map in
```

Comment:
Slightly simpler would be to just have this else-clause, with `let existing_params = try Hashtbl.find ... with Not_found -> []`.